scientific_library/tvb/tests/library/simulator/integrators_test.py

```diff
@@ -179,40 +182,55 @@ def test_clamp(self):
         x = numpy.ones((5, 4, 2))
         for i in range(10):
             x = vode.scheme(x, self._dummy_dfun, 0.0, 0.0, 0.0)
+            vode.bound_and_clamp(x)
         for idx, val in zip(vode.clamped_state_variable_indices, vode.clamped_state_variable_values):
             assert numpy.allclose(x[idx], val)
 
-    def test_update_variables(self):
+    def _test_update_variables(self, model):
+        model.configure()
         x = numpy.ones((5, 4, 2))
         x[:, 0, ] = -x[:, 0, ]
         x[:, 1, ] = 2 * x[:, 1, ]
-        x0 = numpy.array(x)
-        model = TestUpdateVariablesModel()
         for integrator_class in INTEGRATORStoTEST:
             integrator = integrator_class()
             integrator.dt = dt
             try:
                 # Set noise to 0 if this is a stochastic integrator
-                integrator.noise.nsig = numpy.array([0.0])
-                integrator.noise.dt = dt
+                integrator.noise.configure_white(dt, x[model.state_variable_mask].shape)
+                integrator.noise.nsig = numpy.array([0.0] * self.model.nintvar)
             except:
                 pass
+            integrator.configure()
+            integrator.configure_boundaries(model)
+            if model.nvar - model.nintvar:
```

Comment:
this is an error prone thingy that had better be wrapped as a property on the base model class, something like
```python
@property
def has_interm_vars(self):
    return len(model.nvar) > len(model.nintvar)
```

Reply:
> this is an error prone thingy that had better be wrapped as a property on the base model class, something like

I have added

nnonintvar (= _nvar - _nintvar)

and 

has_nonint_vars (= nnonintvar > 0)

properties.

Using has_nonint_vars in the integrators_test now.

Reply:
thanks